### DISCUSSION TITLE
Schedule group `when()` issue

### DISCUSSION THREAD

**Discussion by rcerljenko** [laravel/framework#53608] [Q&A]
### Laravel Version

11.33.2

### PHP Version

8.3

### Description

First of all, I'm not sure if this is a issue with a `when()` function or a `group()` function but it happens after schedule `group()` comes after a `when()`.

Consider this piece of code:
```php
Schedule::daily()
	->group(static function (ConsoleSchedule $schedule): void {
		$schedule->when(static fn (): bool => true)
			->group(static function (ConsoleSchedule $schedule): void {
				// Schedule definitions under "when" condition
			});
		
		// Other daily schedule definitions
	});
```

All schedules should have the same cron expression (`0 0 * * *`) but, as it turns out, the inner group under `when` condition has a cron expression which triggers on every minute (`* * * * *`) and the outer group has the right one.

Something happens with the `when -> group` combo with cron definitions.

### Steps To Reproduce

```php
<?php

// routes/console.php

use Illuminate\Support\Facades\Schedule;
use Illuminate\Console\Scheduling\Schedule as ConsoleSchedule;

Schedule::onOneServer()
	->runInBackground()
	->withoutOverlapping()
	->group(static function (ConsoleSchedule $schedule): void {
		// DAILY
		$schedule->daily()
			->group(static function (ConsoleSchedule $schedule): void {
				$schedule->when(static fn (): bool => true)
					->group(static function (ConsoleSchedule $schedule): void {
						// Schedule definitions under "when" condition
					});
				
				// Other daily schedule definitions
			});
	});
```

**Comment by rcerljenko** [comment]
hi @istiak-tridip, could you take a look please

**Reply by istiak-tridip** [reply]
If I write a nested schedule like this:

```php
Schedule::daily()
    ->group(static function (ConsoleSchedule $schedule): void {
        $schedule->when(static fn (): bool => true)
            ->group(static function (ConsoleSchedule $schedule): void {
                $schedule->command('command-one');
            });

        $schedule->command('command-two');
    });
```
I get output like this from the `schedule:list` command, which is correct:
```
0 0 * * *  php artisan command-one
0 0 * * *  php artisan command-two
```

Am I misunderstanding something 🤔? Can you please share the steps to reproduce or provide an example repo?

**Comment by rcerljenko** [comment]
ok i figured out what's the catch... please reorder the inner and outer commands like so:
```php
Schedule::daily()
    ->group(static function (ConsoleSchedule $schedule): void {
        $schedule->command('command-outer');

        $schedule->when(static fn (): bool => true)
            ->group(static function (ConsoleSchedule $schedule): void {
                $schedule->command('command-inner');
            });
    });
```

**Reply by istiak-tridip** [reply]
I was able to reproduce the issue and will work on debugging and fixing it. Thanks for reporting it!

**Reply by rcerljenko** [reply]
thank you!

**Comment by istiak-tridip** [comment]
Hey @rcerljenko  I've opened a PR #53626 to fix this issue. Please check and provide feedback.

**Reply by rcerljenko** [reply]
thanks man!
